backend/app/tasks/subscription_tasks.py

```
@@ -747,6 +747,28 @@ def check_due_subscriptions(self):
+                    if expires_at_str:
+                        try:
+                            from app.services.subscription.helpers import (
+                                is_subscription_expired,
+                            )
+
+                            expires_at = datetime.fromisoformat(expires_at_str)
+                            if is_subscription_expired(expires_at, now_utc):
+                                # Auto-disable expired subscription
+                                internal["enabled"] = False
+                                sub.json["_internal"] = internal
+                                flag_modified(sub, "json")
```

Ruff (0.15.9)

[error] Undefined name `flag_modified`

(F821)